### PR TITLE
Accommodate changes to XContent classes

### PR DESCRIPTION
**Pull request by msfroh**
There was a commit to OpenSearch core to move XContent classes, forcing us to update a bunch of imports.

This change is effectively a cherry-pick of
87dafb05f3b7893afb5cd0a6efe4674f4aa88c31 without modifying the build.gradle file.
 
### Issues Resolved
https://github.com/opensearch-project/search-processor/issues/120
 
### Check List
- [ ] ~~New functionality includes testing.~~
  - [x] All tests pass
- [ ] ~~New functionality has been documented.~~
  - [ ] ~~New functionality has javadoc added~~
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/search-processor/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java [hunk 1]
```diff
@@ -76,7 +77,7 @@ public void writeTo(StreamOutput out) throws IOException {
     this.properties.writeTo(out);
   }
 
-  public static ResultTransformerConfiguration parse(XContentParser parser) throws IOException {
+  public static KendraIntelligentRankingConfiguration parse(XContentParser parser) throws IOException {
```
Comment (macohen) [inline src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java:80]: re:naming - is this going from less specific naming to more specific intentionally?

File: src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java [hunk 2]
```diff
@@ -154,7 +155,7 @@ public KendraIntelligentRankingProperties(final List<String> bodyFields,
 
     public KendraIntelligentRankingProperties(StreamInput input) throws IOException {
       this.bodyFields = input.readStringList();
-      this.bodyFields = input.readStringList();
+      this.titleFields = input.readStringList();
```
Comment (macohen) [inline src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java:158]: is this related to the XContent classes change?

File: src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java (hunk 1 above)
Comment (msfroh) [reply src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java:80]: Yes.

a) It helps the unit tests, since we can confirm the specific type.
b) There's a best-practice to take the most general type of argument and return the most specific type. I think it's an application of the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle).

File: src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java (hunk 2 above)
Comment (msfroh) [reply src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java:158]: Not directly. 

When merging the cherry-picked commit, I also picked up the unit tests that I had previously added (in KendraIntelligentRankingConfigurationTests).

When I added those tests in `main`, they also revealed some bugs (like this one). I should go through and make sure that all of the tests I added in `main` get backported to 2.x, since I fixed a bunch of little bugs along the way.

Comment (msfroh) [reply src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java:158]: Oh... I just added the `backport 2.x` label to https://github.com/opensearch-project/search-processor/pull/99.

Comment (msfroh) [reply src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java:158]: Oh, but the backport PR fails to build because we need to merge this PR to fix the XContent stuff:

https://github.com/opensearch-project/search-processor/pull/126

**Review by mingshl — APPROVED** [review]

File: src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java (hunk 2 above)
Comment (FredaProMax) [reply src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java:158]: > Oh, but the backport PR fails to build because we need to merge this PR to fix the XContent stuff:
> 
> #126

Looks like there are two ways to handle this issue:
1. In this PR, remove the content of #99, merge this one and merge https://github.com/opensearch-project/search-processor/pull/126
2. Change the description to include the backport of #99, then merge this one; the backport PR https://github.com/opensearch-project/search-processor/pull/126 is pointless in this case because the code is already there.

I'm fine with either, but `1.` appears to overreact, and the order of commits in 2.x is messed up comparing to main branch

Comment (noCharger) [reply src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java:158]: > Oh, but the backport PR fails to build because we need to merge this PR to fix the XContent stuff:
> 
> #126

Looks like there are two ways to handle this issue:
1. In this PR, remove the content of #99, merge this one and merge https://github.com/opensearch-project/search-processor/pull/126
2. Change the description to include the backport of #99, then merge this one; the backport PR https://github.com/opensearch-project/search-processor/pull/126 is pointless in this case because the code is already include.

I'm fine with either, but `1.` appears to overreact, and the order of commits in 2.x is messed up comparing to main branch

Comment (msfroh) [reply src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java:158]: I think #126 shouldn't have any conflicts once this one is merged, because any overlapping changes should be identical.
